src/mender-update/standalone/standalone.cpp

```diff
@@ -201,618 +222,435 @@ error::Error RemoveStateData(database::KeyValueDatabase &db) {
-	err = script_runner.RunScripts(executor::State::ArtifactInstall, executor::Action::Enter);
+	err = state_machine.SetStartStateFromStateData(ctx.state_data.in_state);
 	if (err != error::NoError) {
-		auto install_leave_error {
-			script_runner.RunScripts(executor::State::ArtifactInstall, executor::Action::Error)};
-		log::Error(
-			"Failure during Install Enter script execution: "
-			+ err.FollowedBy(install_leave_error).String());
-		return InstallationFailureHandler(main_context, data, update_module);
+		return {Result::Failed, err};
 	}
 
-	err = update_module.ArtifactInstall();
+	err = state_machine.SetStopAfterState(ctx.stop_after);
 	if (err != error::NoError) {
-		auto install_leave_error {
-			script_runner.RunScripts(executor::State::ArtifactInstall, executor::Action::Error)};
-		log::Error("Installation failed: " + err.FollowedBy(install_leave_error).String());
-		return InstallationFailureHandler(main_context, data, update_module);
+		return {Result::Failed, err};
 	}
 
-	// Install Leave
-	err = script_runner.RunScripts(executor::State::ArtifactInstall, executor::Action::Leave);
-	if (err != error::NoError) {
-		auto install_leave_error {
-			script_runner.RunScripts(executor::State::ArtifactInstall, executor::Action::Error)};
+	state_machine.Run();
 
-		log::Error(
-			"Failure during Install Leave script execution: "
-			+ err.FollowedBy(install_leave_error).String());
-
-		return InstallationFailureHandler(main_context, data, update_module);
-	}
-
-	auto reboot = update_module.NeedsReboot();
-	if (!reboot) {
-		log::Error("Could not query for reboot: " + reboot.error().String());
-		return InstallationFailureHandler(main_context, data, update_module);
-	}
-
-	auto rollback_support = update_module.SupportsRollback();
-	if (!rollback_support) {
-		log::Error("Could not query for rollback support: " + rollback_support.error().String());
-		return InstallationFailureHandler(main_context, data, update_module);
-	}
+	return ctx.result_and_error;
+}
 
-	if (rollback_support.value()) {
-		if (reboot.value() != update_module::RebootAction::No) {
-			return {Result::InstalledRebootRequired, error::NoError};
-		} else {
-			return {Result::Installed, error::NoError};
-		}
+ResultAndError Commit(Context &ctx) {
+	auto exp_in_progress = LoadStateData(ctx.main_context.GetMenderStoreDB());
+	if (!exp_in_progress) {
+		return {Result::Failed, exp_in_progress.error()};
 	}
+	auto &in_progress = exp_in_progress.value();
 
-	cout << "Update Module doesn't support rollback. Committing immediately." << endl;
-
-	auto result = DoCommit(main_context, data, update_module);
-	if (result.result == Result::Committed) {
-		if (reboot.value() != update_module::RebootAction::No) {
-			result.result = Result::InstalledAndCommittedRebootRequired;
-		} else {
-			result.result = Result::InstalledAndCommitted;
-		}
+	if (!in_progress) {
+		return {
+			Result::NoUpdateInProgress,
+			context::MakeError(context::NoUpdateInProgressError, "Cannot commit")};
```

Comment:
similarly here "Cannot commit. No update in progress found"

Reply:
Same.